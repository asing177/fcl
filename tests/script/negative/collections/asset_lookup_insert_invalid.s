global map<int, asset<int>> assMap = ( 3 : a'H1tbrEKWGpbPjSeG856kz2DjViCwMU3qTw3i1PqCLz65' );

transition initial -> terminal;

@initial
swapAssets(decimal<3> amount) {
  transferHoldings(sender(), lookup(3, assMap), amount, sender());
  terminate();
}
