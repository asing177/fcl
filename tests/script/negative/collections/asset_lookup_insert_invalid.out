Error at Line 7, Column 49:
   Implicit loss of numeric precision
      'holdings<asset<int>>' inferred from the type signature of argument 3 of function 'transferHoldings' on Line 7, Column 49
      'decimal<3>' inferred from the type signature of argument 1 of function 'swapAssets' on Line 6, Column 23
   Possible fixes:
      • round the higher precision expression to match the lower precision
      • use types with more (e.g. arbitrary) precision